Classes/BITAuthenticator.m

```diff
@@ -337,17 +337,12 @@ - (void) validateWithCompletion:(void (^)(BOOL validated, NSError *))completion
     return;
   }
   
-  id nsurlsessionClass = NSClassFromString(@"NSURLSessionUploadTask");
-  BOOL isSessionSupported = (nsurlsessionClass && !bit_isRunningInAppExtension());
-  [self validateWithCompletion:completion sessionSupported:isSessionSupported];
-}
-
--(void)validateWithCompletion:(void (^)(BOOL validated, NSError *))completion sessionSupported:(BOOL)isSessionSupported {
```

Comment:
Don't you think it's a good idea to keep the validation in its own method. This will make testing a bit easier.